server/chat-commands.js

```diff
@@ -2803,9 +2805,20 @@ const commands = {
 		let name = this.targetUsername;
 		if (!targetUser && !room.log.hasUsername(target)) return this.errorReply(`User ${target} not found or has no roomlogs.`);
 		let userid = toID(this.inputUsername);
+
 		if (!this.can('mute', null, room)) return;
+		if (targetUser && targetUser.trusted && targetUser !== user) {
+			if (!cmd.includes('force')) {
+				return this.errorReply(`${target} is a trusted user, are you sure you want to hide their messages? Use /forcehidetext if you're sure.`);
+			}
+			// Notify staff when a trusted user gets their messages hidden
+			if (!room.isPrivate) {
+				const staffRoom = Rooms('staff');
+				if (staffRoom) staffRoom.addByUser(user, `<<${room.id}>> ${user.name} hid trusted user ${targetUser.name}'s messages.`);
+			}
+		}
 
-		if (targetUser && (cmd === 'hidealtstext' || cmd === 'hidetextalts' || cmd === 'hidealttext')) {
+		if (targetUser && (cmd.includes('alt'))) {
```

Comment:
```suggestion
		if (targetUser && cmd.includes('alt')) {
```

Reply:
Also, what if you're trying to hide all of the alt's messages of a trusted user? Is this too niche of a situation to worry about?

Reply:
I would say it's too niche to care about